core/alsp_src/tests/libtests/arithx1_tests.pro

```diff
@@ -0,0 +1,131 @@
+
+
+
+ta :- test_arithx1.
+
+test_arithx1 :-
+	test_max,
+	test_min,
+	test_maximum,
+	test_minimum,
+	test_sumlist,
+	test_prodlist,
+	test_sum_squares,
+	test_sum_square_diffs,
+	test_max_vector,
+	test_min_vector.
+
+
+test_max :-
+	max(3,92,M1),
+	M1 == 92,
+	max(3,-92,M2),
+	M2 == 3,
+	max(-6,-9,M3),
+	M3 == -6,
+	max(23.89, 23.87,M4),
+	M4 == 23.89,
+	!.
+	
+test_max :-
+	printf(user, 'test_max test failed\n', []).
+
+
+test_min :-
+	min(3,92,M1),
+	M1 == 3,
+	min(3,-92,M2),
+	M2 == -92,
+	min(-6,-9,M3),
+	M3 == -9,
+	min(23.89, 23.87,M4),
+	M4 == 23.87,
+	!.
+	
+test_min :-
+	printf(user, 'test_min test failed\n', []).
+
+test_maximum :-
+	maximum([3.4, 5, -6.78, 123.4456], M1),
+	M1 == 123.4456,
+	maximum([-3.4, -5, -6.78, -123.4456], M2),
+	M2 == -3.4,
+	!.
+
+test_maximum :-
+	printf(user, 'test_maximum test failed\n', []).
+
+test_minimum :-
+	minimum([3.4, 5, -6.78, 123.4456], M1),
+	M1 == -6.78,
+	minimum([-3.4, -5, -6.78, -123.4456], M2),
+	M2 == -123.4456,
+	!.
+
+test_minimum :-
+	printf(user, 'test_minimum test failed\n', []).
+
+test_sumlist :-
+	sumlist([3.4, 5, -6.78, 123.4456], M1),
+	M1 == 125.0656,
+	sumlist([-3.4, -5, -6.78, -123.4456], M2),
+	M2 == -138.6256,
+	!.
+	
+test_sumlist :-
+	printf(user, 'test_sumlist test failed\n', []).
+
+test_prodlist :-
+	prodlist([3.4, 5, -6.78, 123.4456], M1),
+	R1 is M1 - (- 14228.33986),
+	R1 < 0.0001,
+	prodlist([0.24, 1.6, 8, 23.67654], M2),
+	R2 = M2 - 72.73433088,
```

Comment:
R2 is unified (`=`) here, rather then `is`-calculated like R1 above. Although it doesn't make any difference, it is confusing. In the end it doesn't make any difference. One could avoid the issue by testing with a single query:
```
M2 - 72.73433088 < 0.0001
```

Reply:
Done.